src/compiler/checker.ts

```diff
@@ -7302,6 +7303,12 @@ namespace ts {
                 property.type = typeParameter;
                 properties.push(property);
             }
+            if (strictTuples) {
+                const lengthSymbol = createSymbol(SymbolFlags.Property, "length" as __String);
+                lengthSymbol.type = getLiteralType(arity);
+                lengthSymbol.checkFlags = CheckFlags.Readonly;

```

Comment:
I don't see how adding Readonly helps if the type is literally `2`, because only `2` can be assigned to it. If the intent is future-proofing in case Readonly will someday propagate through assignments, I think that it's as likely to harm as it is to help.

Reply:
Yeah, fair enough. I had no special intentions in adding it. Feel free to adjust as you see fit! :)